File: testing/springboot-testing-demo/src/main/java/com/sivalabs/myservice/repositories/UserRepository.java
package com.sivalabs.myservice.repositories;

import org.springframework.data.jpa.repository.JpaRepository;

import com.sivalabs.myservice.entities.User;
import org.springframework.data.jpa.repository.Query;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long>
{
	@Query("select u from User u where u.email=?1 and u.password=?2")
	Optional<User> login(String email, String password);

    Optional<User> findByEmail(String email);
}

File: spring-boot-k8s-demo/src/main/kotlin/com/sivalabs/geeksclub/users/UserRepository.kt
package com.sivalabs.geeksclub.users

import com.sivalabs.geeksclub.entities.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<User, Long> {
    fun findByEmail(username: String): Optional<User>
}